plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/jdbc/struct/JDBCTableColumn.java

```diff
@@ -200,6 +200,15 @@ public List<DBDLabelValuePair> getValueEnumeration(
         query.append("\nFROM ").append(DBUtils.getObjectFullName(getTable(), DBPEvaluationContext.DML));
         if (valuePattern instanceof String) {
             query.append("\nWHERE ");
+            boolean needCast = false;
+            String typeExt = "";
+            DBSDataType dataType = DBUtils.getDataType(this);
+            if (dataType != null) typeExt = CommonUtils.toString(dataType.geTypeExtension());
+            needCast = "E".equals(typeExt); // for future - maybe not only Enum? But I have no such deep knowledge
+            if (needCast) {
+                identifier = "cast(" + identifier + " as varchar(1000))"; // it could be better for PostgreSQL to specify just "varchar" or "text"
+                                                                          // but this not conforms SQL standard, so I left varchar(1000) - it works  
+            }
```

Comment:
I suppose we don't need to check `"E".equals(typeExt)` if `dataType `is `null`, because it will always be `false`. The same is for cast - we will do nothing here if `dataType `is `null`, so let's add all this under that if statement. Variables declaration are not needed out of this `if`, so they go under `if` too.

Also I'd remove `needCast `variable as it's not so useful for debug purposes as `typeExt`, but makes code more verbose.
```suggestion
            DBSDataType dataType = DBUtils.getDataType(this);
            if (dataType != null) {
                String typeExt = CommonUtils.toString(dataType.geTypeExtension());
                if ("E".equals(typeExt)) { // for future - maybe not only Enum? But I have no such deep knowledge
                    // it could be better for PostgreSQL to specify just "varchar" or "text"
                    // but this not conforms SQL standard, so I left varchar(1000) - it works  
                    identifier = "cast(" + identifier + " as varchar(1000))"; 
            }
```

Reply:
ok, agreed